File: Leetcode/76.cpp
class Solution {
public:
    string minWindow(string s, string t) {
        constexpr int MAXN = 128;
        int m = (int)s.size();
        int stats[MAXN]{};
        for (const char &ch : t) ++stats[ch];
        int frontIndex = 0;
        int stats2[MAXN]{};
        int minLength = numeric_limits<int>::max();
        int lastIndexOfMinLength = -1;
        bool flag;
        for (int i = 0; i < m; ++i) {
            ++stats2[s[i]];
            while (frontIndex <= i && stats2[s[frontIndex]] > stats[s[frontIndex]]) {
                --stats2[s[frontIndex]];
                ++frontIndex;
            }
            flag = true;
            for (char ch = 'a'; ch <= 'z'; ++ch) {
                if (stats2[ch] < stats[ch]) {
                    flag = false;
                    break;
                }
            }
            if (flag) {
                for (char ch = 'A'; ch <= 'Z'; ++ch) {
                    if (stats2[ch] < stats[ch]) {
                        flag = false;
                        break;
                    }
                }
            }
            if (flag && minLength > i-frontIndex+1) {
                minLength = i-frontIndex+1;
                lastIndexOfMinLength = i;
            }
        }
        return (lastIndexOfMinLength == -1) ? string() : string(s.begin()+lastIndexOfMinLength-(minLength-1), s.begin()+lastIndexOfMinLength+1);
    }
};

File: Leetcode/76.go
func minWindow(s string, t string) string {
	const MAXN int = 128
	var stats [MAXN]int
	for _, ch := range t {
		stats[ch]++
	}
	var frontIndex int
	var stats2 [MAXN]int
	var minLength = math.MaxInt32
	var lastIndexOfMinLength = -1
	var flag bool
	for i := 0; i < len(s); i++ {
		stats2[s[i]]++
		for ; frontIndex <= i && stats2[s[frontIndex]] > stats[s[frontIndex]]; frontIndex++ {
			stats2[s[frontIndex]]--
		}
		flag = true
		for ch := 'a'; ch <= 'z'; ch++ {
			if stats2[ch] < stats[ch] {
				flag = false
				break
			}
		}
		if flag {
			for ch := 'A'; ch <= 'Z'; ch++ {
				if stats2[ch] < stats[ch] {
					flag = false
					break
				}
			}
		}
		if flag && minLength > i-frontIndex+1 {
			minLength = i - frontIndex + 1
			lastIndexOfMinLength = i
		}
	}
	if lastIndexOfMinLength == -1 {
		return ""
	}
	return s[lastIndexOfMinLength-minLength+1 : lastIndexOfMinLength+1]
}